python/ray/autoscaler/kubernetes/node_provider.py

```diff
@@ -97,3 +144,25 @@ def get_command_runner(self,
                            docker_config=None):
         return KubernetesCommandRunner(log_prefix, self.namespace, node_id,
                                        auth_config, process_runner)
+
+
+def _add_service_name_to_service_port(spec, svc_name):
+    """Goes recursively through the ingress manifest and adds the
+    right serviceName next to every servicePort definition.
+    """
+    if isinstance(spec, dict):
+        dict_keys = list(spec.keys())
+        for k in dict_keys:
+            spec[k] = _add_service_name_to_service_port(spec[k], svc_name)
+
+            # Every place that contains a servicePort definition should also
+            # have the serviceName defined.
+            if k == "servicePort":
+                spec["serviceName"] = svc_name
```

Comment:
We should probably raise an exception here (or at least print a warning) if the user specified the `serviceName`

Reply:
Good idea! I'll do that!